### ISSUE TITLE
Crash (sigabort) on exit  / zsh history parsing 

### ISSUE THREAD

**Issue by thepaka** [dvorka/hstr#388]
When using hstr with zsh , if you have any entry with more than 15 characters and starting with a colon then hstr will crash on exit. 

```
# echo ': 1579881941:0;:234567890123456' >> ~/.zhistory
# hstr
free(): invalid pointer
zsh: abort (core dumped)  hstr
```

(Archlinux  x86_64 ; zsh 5.7.1-1 ; hstr 2.2-1 )

This might be related to issue https://github.com/dvorka/hstr/issues/353

**Comment by dvorka** [comment]
Great catch!

**Comment by thepaka** [comment]
Fixed by #410
